did-tezos/src/lib.rs

```diff
@@ -60,20 +69,82 @@ impl DIDResolver for DIDTz {
             ..Default::default()
         };
 
-        let doc = Document {
-            context: Contexts::One(Context::URI(DEFAULT_CONTEXT.to_string())),
-            id: did.to_string(),
-            authentication: Some(vec![VerificationMethod::DIDURL(vm_didurl.clone())]),
-            assertion_method: Some(vec![VerificationMethod::DIDURL(vm_didurl.clone())]),
-            verification_method: Some(vec![VerificationMethod::Map(VerificationMethodMap {
-                id: String::from(vm_didurl),
-                type_: proof_type.to_string(),
-                controller: did.to_string(),
-                blockchain_account_id: Some(format!("{}@tezos:{}", address.to_string(), network)),
-                ..Default::default()
-            })]),
-            ..Default::default()
+        let mut doc = DIDTz::tier1_derivation(did, &vm_didurl, proof_type, &address, &network);
+
+        let service = match DIDTz::tier2_resolution(did, &address, &network).await {
+            Ok(s) => s,
+            Err(e) => {
+                return (
+                    ResolutionMetadata {
+                        error: Some(e.to_string()),
+                        ..Default::default()
+                    },
+                    Some(doc),
+                    None,
+                )
+            }
         };
+        doc.service = Some(vec![service]);
+
+        if let Some(s) = &input_metadata.property_set {
+            let public_key = match s.get("public_key") {
+                Some(pk) => match pk {
+                    Metadata::String(pks) => Some(pks.clone()),
+                    _ => {
+                        return (
+                            ResolutionMetadata {
+                                error: Some("Public key is not a string.".to_string()),
+                                ..Default::default()
+                            },
+                            Some(doc),
+                            None,
+                        );
+                    }
+                },
+                None => None,
+            };
+
+            if let Some(updates_metadata) = s.get("updates") {
```

Comment:
Should the key be "signed-ietf-json-patch"? As in the example here:
https://github.com/decentralized-identity/did-spec-extensions/blob/master/parameters/signed-ietf-json-patch.md#examples

Reply:
that’s fine, we can update the spec for this too

Reply:
My thinking is that we (will) have multiple types of updates, but all provided updates have to be of the same type. So from Rust's perspective it looks like this 
```rust
#[derive(Deserialize)]
#[serde(rename_all = "kebab-case")]
#[serde(tag = "type", content = "value")]
enum Updates {
    SignedIetfJsonPatch(Vec<String>),
}
```
> The `String` is a JWS.

And for the Tezos updates I suppose the inner type will also be `Vec<String>` so I thought it made sense to have them under the same key.

Reply:
I think we could consider updating this in a future edition--the different field name would help a resolver differentiate if it's expecting unsigned tx or ietf json patches.